### PR TITLE
Make JsonRpcBase an abstract base type

### PR DESCRIPTION
**Pull request by skynode**
Since `JsonRpcBase` is intended to be a base class, make it impossible for the type to be instantiated anywhere else.

### CODE REVIEW

**Comment by murlokito** [comment]
Makes perfect sense to me. Going to approve and after @tiago18c approval will merge! Thanks for your contribution :)

**Review by murlokito — APPROVED** [review]

**Review by tiago18c — APPROVED** [review]
Looks good. Thanks for the contribution 😁

**Pull request merged**
